### PR TITLE
add 'docker__registry_mirrors' & 'docker__insecure_registries'

### PR DESCRIPTION
**Pull request by harryzorus**

### CODE REVIEW

**Comment by drybjed** [comment]
Is `docker__insecure_registries` really needed? Why not just add the CA certificate of the unknown registry to the host using `debops.pki`? That way the host can authenticate the registry and additional list of 'insecure registries' isn't needed.

If this list is added, I bet that when someone encounters an issue, they will just add that registry to that list instead of fixing the problem in the first place. I would remove that variable and rely on proper PKI security practices instead.

File: templates/etc/default/docker.j2 [hunk 1]
```diff
@@ -40,6 +40,20 @@
 {%     set _ = docker__tpl_options.append("--label " + key + '="' + value + '"') %}
 {%   endfor %}
 {% endif %}
+{% if docker__registry_mirrors|d() %}
+{%   for registry_mirror in docker__insecure_registries %}
```
Comment (drybjed) [inline templates/etc/default/docker.j2]: You use the wrong list variable as the data source, the `for` loop should use the `docker__registry_mirrors` variable instead.

Comment (harryzorus) [inline templates/etc/default/docker.j2]: I have fixed the variable name.

**Comment by harryzorus** [comment]
We should be adding the certificate of a public registry to properly verify the containers we are pulling. The main use case for us is to use an internal registry within our internal network, where encrypting traffic within the network doesn't make much sense.

The variable `docker__insecure_registries` is not absolutely needed, as we can specify the insecure registry using `docker__options`.

**Comment by drybjed** [comment]
If the `docker__insecure_registries` is not needed, then please remove it, since as you said, you can use `docker__options` for that.

**Comment by harryzorus** [comment]
I have removed `docker__insecure_registries` as you suggested.

**Comment by harryzorus** [comment]
closing the the pull request as the requested changes landed upstream through @tallandtree's pull request.

**Pull request closed without merging**